functions.opengraph.php

```diff
@@ -152,16 +152,20 @@ function jetpack_og_tags() {
 			$tags['og:description'] = wp_kses( trim( convert_chars( wptexturize( $tags['og:description'] ) ) ), array() );
 		}
 
-		$tags['article:published_time'] = date( 'c', strtotime( $data->post_date_gmt ) );
-		$tags['article:modified_time']  = date( 'c', strtotime( $data->post_modified_gmt ) );
+		$tags['article:published_time'] = gmdate( 'c', strtotime( $data->post_date_gmt ) );
+		$tags['article:modified_time']  = gmdate( 'c', strtotime( $data->post_modified_gmt ) );
 		if ( post_type_supports( get_post_type( $data ), 'author' ) && isset( $data->post_author ) ) {
 			$publicize_facebook_user = get_post_meta( $data->ID, '_publicize_facebook_user', true );
 			if ( ! empty( $publicize_facebook_user ) ) {
 				$tags['article:author'] = esc_url( $publicize_facebook_user );
 			}
 		}
+	} elseif ( is_search() ) {
+		$query_param = get_query_var( 's', '' );
+		if ( '' !== $query_param ) {
+			$tags['og:title'] = __( 'Search results: ', 'jetpack' ) . $query_param;
+		}
```

Comment:
Looking more into it, I wonder if we could use `wp_get_document_title()` here as well; it does seem to handle search result pages for us already, and we'd get something that's filterable by other plugins:
https://github.com/WordPress/WordPress/blob/master/wp-includes/general-template.php#L1056-L1059

Reply:
Oh nice, I wasn't aware of that method. Let me give it a shot.